### ISSUE TITLE
poor mouselook performance in osx

### ISSUE THREAD

**Issue by jonbro** [sirjuddington/SLADE#253]
In the 3d view of the map editor on osx, there is pretty bad performance for mouselook. WASD is fine, everything there seem to be speedy, but the mouse look only updates at 1fps (though this is highly inconsistant)

**Comment by alexey-lysiuk** [comment]
Mouse look in 3D mode is more or less OK in versions 3.1.0.2 and 3.1.0.3 (built with wxWidgets 3.0.0).
But it's completely broken with the latest wxWidgets (https://github.com/wxWidgets/wxWidgets/commit/39b9a6bbeb1aaff25621b7bbbeaf678bc3974819): camera is constantly spinning and point of view is locked at the ceiling.

**Comment by sirjuddington** [comment]
Typical wxwidgets there >_>

Eventually I should just replace the entire map editor window with an SFML window and custom draw the whole interface (except for dialogs, obviously)

**Comment by parashep** [comment]
Was there ever a solution to this? 3D mode is basically unusable in the OS X binaries, should I build with the older version of wxWidgets?

**Comment by alexey-lysiuk** [comment]
Not really. There three options for wxWidgets to build with:
- 3.0.2 Cocoa backend. The binary was build with it
- 3.0.2 Carbon backend. IIRC it behaved better in 3D mode but had other issues for sure
- 3.1.0 Cocoa backend. Honestly I don't remember was it OK or not in 3D view

If you have a willingness to build SLADE by yourself there are some opportunities.
[Here](https://github.com/alexey-lysiuk/slade-osx-build) is my build environment. `master` branch uses wxWidgets 3.0.2 Cocoa, `wx31` branch is upgraded to 3.1.0 Cocoa. [Here](https://github.com/alexey-lysiuk/SLADE/tree/osx-carbon) is my outdated fork with 3.0.0 Carbon. Headers/libraries can be grabbed from there.
Also you can try to build the latest wxWidgets yourself.

**Comment by parashep** [comment]
Indeed, I built the latest revision of wxWidgets and it still has the problem, just like 3.1.0 and 3.0.2. Your Carbon fork doesn't have the problem.

I tried building the latest version of SLADE with the Carbon libraries, but couldn't using the files from your fork (no wx-config?) and it doesn't seem like it's possible to build wxWidgets Carbon on OS 10.11. Of course, building with Carbon isn't a proper solution anyways.
